### PR TITLE
No longer store auth guard.

### PR DESCRIPTION
**Pull request by garret-gunter**
Fix for #35

### CODE REVIEW

File: src/Blameable/BlameableExtension.php [hunk 1]
```diff
@@ -6,23 +6,17 @@
...
-     * @var Guard
-     */
-    protected $auth;
 
     /**
-     * @param Guard $auth
+     *
      */
-    public function __construct(Guard $auth)
+    public function __construct()
```
Comment (patrickbrouwers) [inline src/Blameable/BlameableExtension.php]: Please remove the unused constructor.

File: src/Loggable/LoggableExtension.php [hunk 2]
```diff
@@ -6,23 +6,17 @@
...
-     * @var Guard
-     */
-    protected $auth;
 
     /**
-     * @param Guard $auth
+     *
      */
-    public function __construct(Guard $auth)
+    public function __construct()
```
Comment (patrickbrouwers) [inline src/Loggable/LoggableExtension.php]: Please remove unused constructor

File: src/ResolveUserDecorator.php [hunk 3]
```diff
@@ -67,8 +59,8 @@ public function setAnnotationReader($reader)
      */
     public function __call($method, $params)
     {
-        if ($this->auth->check()) {
-            call_user_func([$this->wrapped, $this->userSetterMethod], $this->auth->user());
+        if (app('auth')->guard()->check()) {
```
Comment (patrickbrouwers) [inline src/ResolveUserDecorator.php]: Can you add a getter for `getAuth()` (`return app('auth')`) please.

**Review by patrickbrouwers — CHANGES_REQUESTED** [review]

**Comment by garret-gunter** [comment]
Applied the changes.

**Comment by garret-gunter** [comment]
Do you see anything else that should be changed?

**Comment by garret-gunter** [comment]
Is there anything I can do to help this get pulled in?

File: src/ResolveUserDecorator.php [hunk 4]
```diff
@@ -67,8 +59,8 @@ public function setAnnotationReader($reader)
      */
     public function __call($method, $params)
     {
-        if ($this->auth->check()) {
-            call_user_func([$this->wrapped, $this->userSetterMethod], $this->auth->user());
+        if ($this->getAuth()->guard()->check()) {
+            call_user_func([$this->wrapped, $this->userSetterMethod], $this->getAuth());
```
Comment (patrickbrouwers) [inline src/ResolveUserDecorator.php]: This changes behaviour no? We used to pass the logged in user here. Now it's passing the auth manager.

File: src/ResolveUserDecorator.php [hunk 5]
```diff
@@ -96,4 +88,13 @@ public function getEventSubscriberClass()
     {
         return get_class($this->wrapped);
     }
+    
+    /**
+    * Get current Auth manager.
+    *
+    * @return \Illuminate\Auth\AuthManager
+    */
+    protected function getAuth(){
```
Comment (patrickbrouwers) [inline src/ResolveUserDecorator.php]: Can you fix cs here. Brace on new line (see other methods)

File: src/ResolveUserDecorator.php [hunk 6]
```diff
@@ -96,4 +88,13 @@ public function getEventSubscriberClass()
     {
         return get_class($this->wrapped);
     }
+    
+    /**
+    * Get current Auth manager.
+    *
+    * @return \Illuminate\Auth\AuthManager
```
Comment (patrickbrouwers) [inline src/ResolveUserDecorator.php]: `Illuminate\Contracts\Auth\Factory`

**Review by patrickbrouwers — CHANGES_REQUESTED** [review]

File: src/ResolveUserDecorator.php (hunk 4 above)
Comment (garret-gunter) [reply src/ResolveUserDecorator.php]: Yes, it does.  That is a typo.  Good call.

File: src/ResolveUserDecorator.php [hunk 7]
```diff
@@ -96,4 +88,14 @@ public function getEventSubscriberClass()
...
     }
+    
+    /**
+    * Get current Auth manager.
+    *
+    * @return \Illuminate\Contracts\Auth\Factory
+    */
+    protected function getAuth()
+    {
+    	return app('auth');
```
Comment (patrickbrouwers) [inline src/ResolveUserDecorator.php]: Perhaps better to return the guard here straight away.
`->user()` doesn't exist on `\Illuminate\Contracts\Auth\Factory`. (Probably through a magic call, but rather have it explicit)

**Review by patrickbrouwers — CHANGES_REQUESTED** [review]

File: src/ResolveUserDecorator.php (hunk 7 above)
Comment (garret-gunter) [reply src/ResolveUserDecorator.php]: Fixed.  Yes, the default auth manager passes __call() straight to the guard.  I agree that explicit is better.

**Comment by patrickbrouwers** [comment]
Thanks!

**Pull request merged**